### PR TITLE
IA Refactor Pt. 1

### PR DESCRIPTION
**Pull request by tannermpeterson**

### CODE REVIEW

**Review by NikitaPoly — DISMISSED** [review]

File: frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js [hunk 1]
```diff
@@ -396,8 +393,16 @@ export default function InteractiveWaveformModal({
...
+    const { start_time, end_time } = selectedJob.analysisParams;
+    setEditableStartEndTimes({
+      startTime: start_time,
+      endTime: end_time,
+    });
+  };
+
   const loadExistingData = () => {
-    // this happens very fast so not storing to react state the first call, see line 162
+    // this happens very fast so not storing to react state the first call, see line 162 (? different line now)
```
Comment (luciipak) [inline frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js:405]: did I write that comment? I don't really know what I meant by that

File: frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js [hunk 2]
```diff
@@ -182,67 +182,104 @@ const contextMenuItems = {
...
   const [menuItems, setMenuItems] = useState(contextMenuItems.moveDelete);
   const [selectedMarkerToMove, setSelectedMarkerToMove] = useState();
   const [cursorLoc, setCursorLoc] = useState([0, 0]);
   const [xZoomFactor, setXZoomFactor] = useState(1);
   const [yZoomFactor, setYZoomFactor] = useState(1);
-  /* NOTE!! The order of the variables and functions in createGraph() are important to functionality.
+
+  // these functions are just wrappers around hook setters, so adding them before useEffects
+  const updateFeatures = (featureType, newFeatureValues) => {
+    console.log(featureType, newFeatureValues);
```
Comment (luciipak) [inline frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js:220]: console.log

File: frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js [hunk 3]
```diff
@@ -182,67 +182,104 @@ const contextMenuItems = {
...
+    // ensures you don't edit the original array by creating deep copy
+    const newEntries = JSON.parse(JSON.stringify(editablePeaksValleys));
+    const featureIdx = featureType === "peak" ? 0 : 1;
+    newEntries[selectedWell][featureIdx] = newFeatureValues;
+    setEditablePeaksValleys(newEntries);
+  };
+
+  const updateStartEndTimes = (newStartEndTimes) => {
```
Comment (luciipak) [inline frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js:228]: these are great, goes off that article, I"m happy to incorporate this more

File: frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js [hunk 4]
```diff
@@ -953,29 +906,12 @@ export default function WaveformGraph({
...
-        const idxVal = peaks[idx];
-        peaks.splice(idx, 1, (idxVal += 1));
-        setPeaks([...peaks]); // required to change dependencies
-      } else {
-        const idxVal = valleys[idx];
-        valleys.splice(idx, 1, (idxVal += 1));
-        setValleys([...valleys]); // required to change dependencies
-      }
-    }
+    // Tanner (5/25/23): Currently assuming that only 37 and 39 will be passed in
```
Comment (luciipak) [inline frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js:909]: What happens if another key is typed?

Comment (tannermpeterson) [reply frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js:909]: I'll have to check

File: frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js (hunk 1 above)
Comment (tannermpeterson) [reply frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js:405]: I'm not sure who wrote it

File: frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js (hunk 4 above)
Comment (tannermpeterson) [reply frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js:909]: Should only call this function though if one of those two keys are pressed https://github.com/CuriBio/k8s_curi/pull/537/files/4b7c9ae835d7cecdeb83d2b1f2c9f72151e39786#diff-fc4024e23c0ac1f90740005d341ae9dbd8db7fdbfac7ce3c5cba660f2c97f59dR364

**Review by NikitaPoly — APPROVED** [review]

**Review by luciipak — APPROVED** [review]

**Pull request merged**